source/language_service/src/completion/tests.rs

```diff
@@ -2379,3 +2380,170 @@ fn in_trailing_comment() {
         }",
     );
 }
+
+#[test]
+fn export_from_dependency_in_scope() {
+    check_with_dependency(
+        r"
+        namespace Test {
+            open MyDep;
+            operation Foo() : Unit {
+                ↘
+            }
+        }
+        ",
+        "MyDep",
+        "
+        namespace Bar {
+            operation Baz() : Unit {}
+            export Baz;
+        }
+        namespace Main {
+            operation Qux() : Unit {}
+            export Qux, Bar.Baz;
+        }
+        ",
+        &["Qux", "Baz", "Bar"],
+        &expect![[r#"
+            found, sorted:
+              "Baz" (Function)
+                detail: "operation Baz() : Unit"
+              "Qux" (Function)
+                detail: "operation Qux() : Unit"
+              "Bar" (Module)
+        "#]],
+    );
+}
+
+#[test]
+fn aliased_export_from_dependency_in_scope() {
+    check_with_dependency(
+        r"
+        namespace Test {
+            open MyDep;
+            operation Foo() : Unit {
+                ↘
+            }
+        }
+        ",
+        "MyDep",
+        "
+        namespace Bar {
+            operation Baz() : Unit {}
+            export Baz;
+        }
+        namespace Main {
+            export Bar.Baz, Bar.Baz as BazAlias;
+        }
+        ",
+        &["BazAlias", "Baz"],
+        &expect![[r#"
+            found, sorted:
+              "Baz" (Function)
+                detail: "operation Baz() : Unit"
+              "BazAlias" (Function)
+                detail: "operation Baz() : Unit"
+        "#]],
+    );
+}
+
+#[test]
+fn namespace_export_from_dependency_qualified() {
+    check_with_dependency(
+        r"
+        namespace Test {
+            open MyDep.Baz.↘
+        }",
+        "MyDep",
+        "namespace Foo.Bar {
+            operation Qux() : Unit {}
+            export Qux
+         }
+         namespace Baz {
+            export Foo.Bar;
+         }",
+        &["Bar"],
+        &expect![[r#"
+            found, sorted:
+              "Bar" (Module)
+        "#]],
+    );
+}
+
+#[test]
+fn export_from_dependency_qualified() {
+    check_with_dependency(
+        r"
+            namespace Test {
+                operation Test() : Unit {
+                    MyDep.↘
+                }
+            }",
+        "MyDep",
+        "namespace Foo {
+                operation Baz() : Unit {}
+                export Baz;
```

Comment:
I know we've talked before about this, but it is really confusing to me. Do we need to export `Baz` here if it is already exported in `Main` as `Foo.Baz`?

Reply:
This is kind of like how in Rust:

```rs
pub mod foo {
   pub use other_mod::bar;
}
```

Would create `foo::bar`, _not_ `foo::other_mod::bar`.

Reply:
Thanks for the explanation!